controllers/cloud.redhat.com/errors/errors.go

```diff
@@ -73,22 +73,38 @@ func Wrap(msg string, err error) *ClowderError {
 	return clowderErr
 }
 
-// MissingDependencies is a struct containing a map of string lists, it is intended to hold
-// information about missing dependencies. The key is a resource type, or a provider as a string,
-// and the list is a list of keys or items that are missing.
+//MissingDependency is a struct that holds information about a missing dependency
+type MissingDependency struct {
+	Source  string
+	App     string
```

Comment:
Is the app strictly necessary here? since we will always be throwing the error in the context of a specific application, and its place in the status field places it inside of an application, is the app name here redundant?

Reply:
I believe it is because the MissingDependency struct provides its own ToString method for producing the error message which is derives from the source, app, and details. This ensures the messages are consistent, and that this logic is only implemented once.

Reply:
I thought the app was already present in the log message by virtue f the fact that it was being emitted from the app itself. If so, I see no need to pass it in here.

Reply:
Oh! I get what you are saying now. I totally didn't understand. Because the log call is made in the context of an app the app name will be there automatically and our missing dep error is added to it. That makes sense. I thought you were asking me to interpolate the app name into the error string within each provider, which would have been messy. Removed app.